main.tf

```diff
@@ -66,6 +72,11 @@ resource "google_pubsub_subscription" "pull_subscriptions" {
     "ack_deadline_seconds",
     local.default_ack_deadline_seconds,
   )
+  message_retention_duration = lookup(
```

Comment:
Instead of setting a default duration ourselves, we should simply set this field to `null` if it's not configured.

Reply:
Makes sense. I was blithely following the pattern used for `ack_deadline_seconds`. One mod coming up..